ExamenPGG.Business/Squares/Factory/SquareFactory.cs

```diff
@@ -7,16 +9,31 @@ public ISquare CreateSquare(int id, SquareType SquareType)
             switch (SquareType)
             {
                 case SquareType.Start:
-                    return new StartSquare(id);
+                    return new Start(id);
```

Comment:
maybe its better to inject a logger into each square, rather than to create a logger inside each square.